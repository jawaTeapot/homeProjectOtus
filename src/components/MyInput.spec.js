import { describe, beforeEach, expect, test } from "vitest";
import component from "./MyInput.vue";
import { mount } from "@vue/test-utils";

describe("MyInput component", () => {
  function buildWrapper(value) {
    const wrapper = mount(component, {
      propsData: { modelValue: value },
      components: {
        ElInput: {
          template: `<input :value="value" @change="$emit('update:modelValue', $event.target.value)">`,
          props: ['value', 'size', 'prefix-icon'],
        },
      },
    });
    return wrapper;
  }
  beforeEach(async () => {});

  test("mounts without errors", () => {
    const wrapper = buildWrapper("hello search");
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("can input", () => {
    const wrapper = buildWrapper('hello search');
    const input = wrapper.find('input')
    input.setValue('updated!')
    input.trigger('change')
    expect(wrapper.emitted()['update:modelValue'][0]).toEqual(['updated!'])
  });
});
