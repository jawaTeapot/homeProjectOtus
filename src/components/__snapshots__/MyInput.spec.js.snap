// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyInput component > mounts without errors 1`] = `"<input modelvalue="hello search" placeholder="Поиск ..." class="w-[180px] ml-3">"`;
